x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/BlobCacheMetrics.java

```diff
@@ -56,4 +102,47 @@ public LongCounter getEvictedCountNonZeroFrequency() {
     public LongHistogram getCacheMissLoadTimes() {
         return cacheMissLoadTimes;
     }
+
+    /**
+     * Record the various cache population metrics after a chunk is copied to the cache
+     *
+     * @param totalBytesRead The total number of bytes read
+     * @param totalReadTimeNanos The time taken to read the bytes in nanoseconds
+     * @param totalBytesWritten The total number of bytes written
+     * @param totalWriteTimeNanos The time taken to write the bytes in nanoseconds
+     * @param elapsedTimeNanoseconds The time taken to copy the entire chunk in nanoseconds
+     * @param shardId The shard ID to which the chunk belonged
+     * @param cachePopulationReason The reason for the cache being populated
+     */
+    public void recordCachePopulationMetrics(
+        int totalBytesRead,
+        long totalReadTimeNanos,
+        int totalBytesWritten,
+        long totalWriteTimeNanos,
+        long elapsedTimeNanoseconds,
+        ShardId shardId,
+        CachePopulationReason cachePopulationReason
+    ) {
+        Map<String, Object> metricAttributes = Map.of(
+            SHARD_ID_ATTRIBUTE_KEY,
+            shardId,
+            CACHE_POPULATION_REASON_ATTRIBUTE_KEY,
+            cachePopulationReason
+        );
+        cachePopulateReadThroughput.record(toBytesPerSecond(totalBytesRead, totalReadTimeNanos), metricAttributes);
+        cachePopulateWriteThroughput.record(toBytesPerSecond(totalBytesWritten, totalWriteTimeNanos), metricAttributes);
+        cachePopulationElapsedTime.record(TimeUnit.NANOSECONDS.toMillis(elapsedTimeNanoseconds), metricAttributes);
+    }
+
+    /**
+     * Calculate throughput as bytes/second
+     *
+     * @param totalBytes The total number of bytes transferred
+     * @param totalNanoseconds The time to transfer in nanoseconds
+     * @return The throughput as bytes/second
+     */
+    private int toBytesPerSecond(int totalBytes, long totalNanoseconds) {
+        double totalSeconds = totalNanoseconds / 1_000_000_000.0;
+        return (int) (totalBytes / totalSeconds);
+    }
```

Comment:
We could do this as integer arithmetic e.g.
```java
return (int) ((totalBytes * 1_000_000_000L) / totalNanoseconds);
```
But then we have to worry about overflow and/or precision

Reply:
If the approach is deemed sound I'll protect against divide-by-zero here (nano timestamps can be very coarse on some platforms)